pkg/tools/cli/loadtest/service/search/search.go

```diff
@@ -0,0 +1,89 @@
+//
+// Copyright (C) 2019-2020 Vdaas.org Vald team ( kpango, rinx, kmrmt )
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+package search
+
+import (
+	"context"
+	"reflect"
+	"sync"
+
+	"github.com/vdaas/vald/internal/client"
+	"github.com/vdaas/vald/internal/errgroup"
+	"github.com/vdaas/vald/internal/errors"
+	"github.com/vdaas/vald/internal/safety"
+	"github.com/vdaas/vald/pkg/tools/cli/loadtest/assets"
+)
+
+type search struct {
+	eg  errgroup.Group
+	r   client.Reader
+	c   int
+	n   string
+	req []*client.SearchRequest
+}
+
+func New(opts ...SearchOption) (s *search, err error) {
+	s = new(search)
+	for _, opt := range append(defaultSearchOpts, opts...) {
+		if err = opt(s); err != nil {
+			return nil, errors.ErrOptionFailed(err, reflect.ValueOf(opt))
+		}
+	}
+	s.eg = errgroup.Get()
```

Comment:
It may be better to set it with FOP.:thinking:

Reply:
same above

Reply:
ok! I think it is good.